src/compiler/checker.ts

```diff
@@ -18026,25 +18037,25 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function getFreshTypeOfLiteralType(type: Type): Type {
-        if (type.flags & TypeFlags.Literal) {
-            if (!(type as LiteralType).freshType) {
+        if (type.flags & TypeFlags.Freshable) {
+            if (!(type as FreshableType).freshType) {
                 const freshType = createLiteralType(type.flags, (type as LiteralType).value, (type as LiteralType).symbol, type as LiteralType);
                 freshType.freshType = freshType;
-                (type as LiteralType).freshType = freshType;
+                (type as FreshableType).freshType = freshType;
             }
-            return (type as LiteralType).freshType;
+            return (type as FreshableType).freshType;
         }
         return type;
     }
 
     function getRegularTypeOfLiteralType(type: Type): Type {
-        return type.flags & TypeFlags.Literal ? (type as LiteralType).regularType :
+        return type.flags & TypeFlags.Freshable ? (type as FreshableType).regularType :
             type.flags & TypeFlags.Union ? ((type as UnionType).regularType || ((type as UnionType).regularType = mapType(type, getRegularTypeOfLiteralType) as UnionType)) :
             type;
     }
 
     function isFreshLiteralType(type: Type) {

```

Comment:
perhaps this "family" of functions could be renamed now? they all (`isFreshLiteralType`, `getRegularTypeOfLiteralType`, `getFreshTypeOfLiteralType`, `getRegularTypeOfLiteralType`) mention "literal" type but now they will also handle enums and enums are not literal types

Reply:
Yeah, but `isFreshFreshableType` and `getFreshTypeOfFreshableType` are pretty awful. 😮

Reply:
Yeah, can't argue that 😅 OTOH, those names would be more correct - they sound a little bit weird at first but if we focus on their meaning, it makes sense.

Reply:
The fresh type of bel-air!

I suppose it could be argued that `Enum.A` is an “enum literal”, even if it’s not a literal in the grammatical sense.  The compiler *is* treating it as one in a way by giving it its own nominal type (`Enum`)

Reply:
The main problem from my PoV, is that "literal" is already used in the codebase and it often doesn't include enums:
https://github.com/microsoft/TypeScript/blob/30993855fd418a116ad92e821f7b6d4e52a3ca60/src/compiler/types.ts#L6042

I see now though that `EnumLiteral` also exist so maybe it isn't a big deal.